test/validation/api/atomic/atomic.c

```diff
@@ -1552,6 +1634,11 @@ static void atomic_test_atomic_max(void)
 	test_atomic_functional(test_atomic_max_thread, test_atomic_validate_max);
 }
 
+static void atomic_test_atomic_fetch_max(void)
+{
+	test_atomic_functional(test_atomic_fetch_max_thread, test_atomic_validate_max);
```

Comment:
The existing test_atomic_validate_max does not validate the atomicity of fetch+max operation at alll. If I replace odp_atomic_fetch_max_u32() in the test function with odp_atomic_load_u32() followed by odp_atomic_max_u32(), the test still passes.

One way to improve this would be to keep track of how much the value of the atomic variable gets increased (i.e. if (tmp > val) increase += tmp - val) and then checking at validation phase that the sum of the thread-specific increases is consistent with the final value of the atomic variable. This would at least catchs the non-atomic load+max pair that I tried.

Reply:
Improved test in V3.